c7n/utils.py

```diff
@@ -967,6 +967,23 @@ def jmespath_search(*args, **kwargs):
     )
 
 
+def get_path(path: str, resource: dict):
+    """
+    This function provides a wrapper to obtain a value from a resource
+    in an efficient manner.
+    jmespath_search is expensive and it's rarely the case that
+    there is a path in the id field, therefore this wrapper is an optimisation.
```

Comment:
It's a convenient side effect of @thisisshi 's work in #8533 that we call custodian's own `jmespath_search()` rather than directly using `jmespath.search()`. So if we're going to optimize away unnecessary calls, we handle that inside `jmespath_search()` which is already the most common pattern for extracting things that might be JMESPath expressions or simple top-level field keys.

(That could definitely be handled in a separate PR if we want to keep this one more focused, but this one is already operating at a pretty high/broad level in many of its changes)

Reply:
Hi I only did this because I a previous commit I was told that the Search was expensive. Welcome better alternative that allows a path.

Reply:
Yeah thanks. Re the request for specificity. The earlier commit didnt include the core changes I made in the final commit.  So there's a leap there. Apols

Reply:
@ajkerrigan should switch to jmespath_search() and optimise it?

Reply:
@ajkerrigan - hi do you want me to do something  with jmespath_search as you suggested?
Thanks John

Reply:
No, thanks

Reply:
I took a look at trying what you suggested however I am doubtful about the approach an my ability to make it float.

I thought this naive approach would work ....

`def jmespath_search(*args, **kwargs):
    # DROP OUT IF ITS A FIELD 
    path = args[0]
    resource = args[1]
    if '.' not in path:
        return resource[path]

    # Otherwise it's a path...
    return jmespath.search(
        *args,
        **kwargs,
        options=jmespath.Options(custom_functions=C7NJmespathFunctions())
    )
`

But, the 3rd options passed to the jmespath.search mean my shortcut might interfer with existing usage if people are already passing simple fields to it.

Also I cam across cases where there were more than 2 args being passed to the wrapper "jmespath_search" meaning again I am uncertain whether my shortcut interfers with existing usage.

Any ideas?  (BTW I am not a python programmer and not even sure how *args and *kwargs actually work or why one would need them)